### PR TITLE
fix1308

### PR DESCRIPTION
**Pull request by linghuiluo**
Address an interesting abnormal case mentioned here https://github.com/soot-oss/soot/pull/1308
A concrete method with no instruction would cause AsmMethodSource throw null point exception.

### CODE REVIEW

**Comment by linghuiluo** [comment]
@mbenz89 Maybe you can have a look at this PR, since you looked into #1308 last year.

**Comment by linghuiluo** [comment]
@swissiety can you review this ?

File: src/main/java/soot/PackManager.java [hunk 1]
```diff
@@ -958,13 +958,11 @@ private void runBodyPacks(SootClass c) {
         // whole shimple or not?
         {
           Body body = m.retrieveActiveBody();
-          if (body instanceof ShimpleBody) {
+          if (body != null && body instanceof ShimpleBody) {
```
Comment (swissiety) [inline src/main/java/soot/PackManager.java:961]: `!= null` and `null instanceof whatever` is redundant as `null instanceof` resolves to false

**Comment by linghuiluo** [comment]
@linghuiluo  I am working on this

**Pull request closed without merging**